monitoring/monitoring/utils.py

```diff
@@ -0,0 +1,165 @@
+import numpy as np
+import pandas as pd
+import plotly.express as px
```

Comment:
Adding a general comment for the `utils` file here. First, have a look at the main comment of the review ;)

I would prefer to split the functions and create the plots in one function and the dataframes in another, since those are often not equivalent.

Reply:
Sure, I can regroup this. But how could I omit or postpone the API call for the dataframes if I need them first to then create the plots?

Reply:
I removed some redundant dataframes, grouped all the charts in the same container and added st.button() to display additional dataframes just if user clicks the button.